middleware/cors.go

```diff
@@ -92,11 +92,9 @@ func (c *CORSConfig) HeadersSupported(requestHeaderFieldNames []string) bool {
 	return true
 }
 
-// WithCORSConfig is middleware that handles your CORS preflight requests quickly
-// and effectively with the supplied configuration. It is not verbose. To enable
-// verbosity, please wrap it with some sort of request logging middleware.
-func WithCORSConfig(next http.Handler, c CORSConfig) http.Handler {
-	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+// Adds CORS functionality to a function, primarily ServeHTTP functions
+func WithCORSConfig(serveFn func(http.ResponseWriter, *http.Request), c CORSConfig) func(http.ResponseWriter, *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Exported functions should have comments before their declaration that begin with their name, i.e. `// WithCORSConfig adds CORS functionality...`.